src/Npgsql/PoolingDataSource.cs

```diff
@@ -76,15 +74,12 @@ internal sealed override (int Total, int Idle, int Busy) Statistics
 
     internal PoolingDataSource(
         NpgsqlConnectionStringBuilder settings,
-        NpgsqlDataSourceConfiguration dataSourceConfig,
-        NpgsqlMultiHostDataSource? parentPool = null)
+        NpgsqlDataSourceConfiguration dataSourceConfig)
         : base(settings, dataSourceConfig)
     {
         if (settings.MaxPoolSize < settings.MinPoolSize)
             throw new ArgumentException($"Connection can't have 'Max Pool Size' {settings.MaxPoolSize} under 'Min Pool Size' {settings.MinPoolSize}");
 
-        _parentPool = parentPool;
```

Comment:
Huh, funny that the only reason we had _parentPool was for this TransactionScope thing...

Reply:
Not only that, at some point there was this weird thing where whenever we open a physical connection we also query cluster's state depending on whether we have a parent pool. No idea why it was done so...